### ISSUE TITLE
Documentation: Mixed/inconsistent  case paths on MacOS cause dreadful problems

### ISSUE THREAD

**Issue by WorikQCI** [flutter/flutter#114040]
tl;dr Can the documentation for installing on MacOS have the advice to use only lower case letters in file paths?

## Steps to Reproduce

We used mixed case in our file paths when we started our project.

`lib/src/data/Enums/some_file.dart`  Some times that path was referred to as: 'lib/src/data/enums/some_file.dart'

MacOS treats "..../Abcd/" as "...../abcd/" in most cases.  But not all cases, for all software, it turns out.

We started having problems of the form:

```
A value of type 'OurDartStructure (where OurDartStructure is defined in <path to project>/lib/src/data/Enums/some_file.dart)' can't be assigned to a variable of type 'OurDartStructure (where OurDartStructure is defined in <path to project>/lib/src/data/enums/some_file.dart)'.
```

We have been through our code, git repositories, file system and made every path element and file name lower case.  Ditto all import paths.

This is no longer a problem but what we learnt should be in the documentation: "On MacOS ensure all path elements are lowewr case/consistent case"

On MacOS the treatment of case in file names causes problems with flutter, but also git (another, similar, horrid saga)

**Comment by stuartmorgan** [comment]
> MacOS treats "..../Abcd/" as "...../abcd/" in most cases.

This isn't a function of macOS, but of the filesystem; macOS has full support for case-sensitive filesystems, it's just not the default format on Mac hardware. The situation is similar on Windows.

> But not all cases, for all software, it turns out.

Whether specific software (such as Dart in the case of the issues you had) treats paths as case-insensitive depends on the software, not macOS.

> On MacOS the treatment of case in file names causes problems with flutter

Can you elaborate on why you want this documented as a macOS-specific issue? On a case-sensitive filesystems the code would have been wrong as well because your import would have been of a file that wouldn't even exist.

It's also not clear that the installation instructions would be the right place for documenting this, since it's about projects, not Flutter installation. And it's already covered in more general docs: https://dart.dev/guides/language/effective-dart/style#do-name-libraries-and-source-files-using-lowercase_with_underscores

**Comment by WorikQCI** [comment]
```
Can you elaborate on why you want this documented as a macOS-specific issue? On a case-sensitive filesystems the code would have been wrong as well because your import would have been of a file that wouldn't even exist.
```

Good point.

The advice should be "use lower case" or "use consistent case" for all paths and file names.  Regardless of OS.

Having spent 30 years exclusively with case sensitive file systems I see "foo/Bar/" as different from "foo/bar" with my mind's eye.  On MacOS (and Windows) that is not guaranteed.

**Comment by stuartmorgan** [comment]
> The advice should be "use lower case" or "use consistent case" for all paths and file names.

As I linked above, that is in fact already the advice for Dart code. What specific change are you looking for?

**Comment by WorikQCI** [comment]
> As I linked above, that is in fact already the advice for Dart code. What specific change are you looking for?

That is that then.  

I would like you to fix MacOS and Git so they play nice together on systems with mixed  case file names....  A bit outside your remit.

**Comment by stuartmorgan** [comment]
As I mentioned above, if you want case-sensitive file handling on macOS all the support is already there. You just have to format the drive containing those files a with a case-sensitive filesystem.

Git also supports case-sensitive file handling. You just can't check out a git repository that requires it (e.g., by having two files that differ only in case) on a case-insensitive filesystem.

**Comment by WorikQCI** [comment]
Mēh!  

Almost anything is possible.

Moving from the Linux world and file systems to Mac ones is a frightening journey because the handling of case in file paths is *inconsistent*.

Since anything is possible every problem you have is your fault.  That is unhelpful, true, but unhelpful.

The solution for cross platform development (I have learnt for myself) is to use consistent case for every path element.  

Lower case is the easiest thing to do and simply avoids all these problems.

FLW
